tests/unit/test_expression_tree/test_operations/test_convert_to_casadi.py

```diff
@@ -205,7 +205,7 @@ def test_interpolation(self):
         data4 = 2 * x4  # np.tile(2 * x3, (10, 1)).T
         with pytest.raises(ValueError, match="Invalid dimension of x"):
             interp = pybamm.Interpolant(x4_, data4, y4, interpolator="linear")
-            interp_casadi = interp.to_casadi(y=casadi_y)
+        interp_casadi = interp.to_casadi(y=casadi_y)
 
```

Comment:
This part looks off to me. If the raise value error happens in the `pybamm.Interpolant()` line then the `to_casadi()` line does nothing